### PR TITLE
feat(examples): refactor `api-routes-graphql` to GraphQL Yoga

### PR DESCRIPTION
**Pull request by charlypoly**
GraphQL Yoga provides a more extensible experience ([Envelop plugins](https://envelop.dev/)) and lighter bundle than the previously showcased `apollo-server-micro` ([64.5kB](https://bundlephobia.com/package/@graphql-yoga/node@2.2.1) vs [196.2kB](https://bundlephobia.com/package/apollo-server-micro@3.6.7) - min & GZiped)

Similar to Next.js, GraphQL Yoga comes with strong defaults and out-of-the-box support for modern GraphQL features (File uploads, subscriptions, and more).

Strong of many years of existence and a large community, GraphQL Yoga 2.0 comes with Next.js support with no additional package or configuration 📦

### CODE REVIEW

**Review by dotansimha — DISMISSED** [review]

File: examples/api-routes-graphql/README.md [hunk 1]
```diff
@@ -1,6 +1,24 @@
 # API routes with GraphQL server
 
-Next.js ships with [API routes](https://nextjs.org/docs/api-routes/introduction), which provide an easy solution to build your own `API`. This example shows their usage alongside [apollo-server-micro](https://github.com/apollographql/apollo-server/tree/main/packages/apollo-server-micro) to provide simple GraphQL server consumed by Next.js app.
+Next.js ships with [API routes](https://nextjs.org/docs/api-routes/introduction), which provide an easy solution to build your own `API`.
```
Comment (leerob) [inline examples/api-routes-graphql/README.md:3]: Is there a deployed example of this we can see?

Comment (charlypoly) [reply examples/api-routes-graphql/README.md:3]: Sure, you'll find this branch deployed on Vercel here: https://api-routes-graphql-yoga.vercel.app/api/graphql?query=%7B%0A++greetings%0A%7D

**Review by ardatan — DISMISSED** [review]

**Review by dotansimha — DISMISSED** [review]

File: examples/api-routes-graphql/package.json [hunk 2]
```diff
@@ -6,12 +6,14 @@
...
-    "micro": "^9.3.4",
+    "@graphql-yoga/node": "^2.2.1",
+    "graphql": "^16.3.0",
     "next": "latest",
     "react": "^17.0.2",
     "react-dom": "^17.0.2",
     "swr": "^0.5.6"
+  },
+  "resolutions": {
+    "graphql": "16.3.0"
```
Comment (ijjk) [inline examples/api-routes-graphql/package.json:17]: Is this resolution needed? Ideally we don't need to include this as it won't apply for `npm`

Comment (charlypoly) [reply examples/api-routes-graphql/package.json]: @ijjk Just tested without the resolutions and pushed a commit to remove it

**Review by ijjk — DISMISSED** [review]
Thanks!

**Review by ijjk — APPROVED** [review]

**Pull request merged**